svir/dialogs/viewer_dock.py

```diff
@@ -940,6 +948,28 @@ def load_avg_losses_rlzs_aggr(
 
         self.filter_avg_losses_rlzs_aggr()
 
+    def get_total_loss_unit(self, loss_type):
+        unit = None
+        separate_types = loss_type.split('+')
+        for separate_type in separate_types:
+            if separate_type in self.single_loss_types:
+                loss_type_idx = self.loss_type_cbx.findText(separate_type)
+                this_unit = self.agg_curves['units'][loss_type_idx]
+                if unit is not None and unit != this_unit:
+                    raise ValueError(
+                        f'The unit of the total loss type ({loss_type}) can'
+                        f' not be determined by the separate loss types'
+                        f' ({separate_types}) because they have different'
+                        f' units ({unit} vs {this_unit})')
```

Comment:
It makes little sense to do this check QGIS side, it is too late: if the units are inconsistent it must be an error engine side, even before starting the calculation.